Adding python generated ecflow definition file with run type options

AnnaSmoot-NOAA opened the pull request:
## DESCRIPTION OF CHANGES: 
<!-- One or more paragraphs describing the problem, solution, and required changes. -->
Currently there are the aqm_cycled.def and aqm_testing.def files which are hard coded. The new ecf_generate.py script generates both of those definition files based on nco_aqm.yml and test_aqm.yml. There is also a third run type user_aqm.yml which allows for more flexible configuration. The Python script also adds machine configurations for C6, and WCOSS2. The run type yaml files are used based on run type set in user_settings.sh. Additionally, the following files are moved to ush/: 
- detect_machine.py
- run_workflow.sh
- setup_workflow.sh
- setup_ecflow.sh
- user_settings.sh
- GETTING_STARTED.md (There are two of these, one in package home and one in ush/)

### Type of change
<!-- Please delete options that are not relevant. Add an X to check off a box. -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update
- [ ] Scientific results will change based on this work 

## TESTS CONDUCTED: 
<!-- Explicitly state what tests were run on these changes, or if any are still pending (for README or other text-only changes, just put "None required"). Make note of the compilers used, the platform/machine, and other relevant details as necessary. For more complicated changes, or those resulting in scientific changes, please be explicit! -->
<!-- Add an X to check off a box. -->

- [X] gaea.intel
- [] hera.intel
- [X] wcoss2.intel
Part of this work has been tested and successful on C6 and WCOSS2. Another round of testing is necessary.

## DEPENDENCIES:
<!-- Add any links to external PRs (e.g. regional_workflow and/or UFS PRs). For example:
- ufs-community/regional_workflow/pull/<pr_number>
- ufs-community/UFS_UTILS/pull/<pr_number>
- ufs-community/ufs-weather-model/pull/<pr_number> -->

## DOCUMENTATION:
<!-- If this PR is contributing new capabilities that need to be documented, please also include updates to the RST files (docs/UsersGuide/source) as supporting material. -->

## ISSUE: 
<!-- If this PR is resolving or referencing one or more issues, in this repository or elsewhere, list them here (Remember, issues must always be created before starting work on a PR branch!). For example, "Fixes issue mentioned in #123" or "Related to bug in https://github.com/ufs-community/other_repository/pull/63" -->

## CHECKLIST
<!-- Add an X to check off a box. -->
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [ ] My changes do not require updates to the documentation (explain).
- [X] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] Any dependent changes have been merged and published

## PRIOTIRY LEVEL
<!-- Add an X to check off a box. -->
- [ ] Critical: emergency fix for a major issue
- [ ] High: upcoming production fix/change 
- [X] Normal: important but not high priority
- [ ] Low: routine addition or bonus feature that can wait

## CONTRIBUTORS (optional): 
<!-- If others have contributed to this work aside from the PR author, list them here -->
Brian Curtis

AnnaSmoot-NOAA committed:
Create PULL_REQUEST_TEMPLATE

AnnaSmoot-NOAA committed:
Create bug_report.md

AnnaSmoot-NOAA committed:
Create feature_request.md

AnnaSmoot-NOAA committed:
Create textonly_request.md

AnnaSmoot-NOAA committed:
Delete .github directory

AnnaSmoot-NOAA committed:
Merge branch 'develop' of https://github.com/NOAA-EMC/AQM-Workflow into develop

AnnaSmoot-NOAA committed:
Merge branch 'develop' of https://github.com/NOAA-EMC/AQM-Workflow into develop

AnnaSmoot-NOAA committed:
Merge branch 'develop' of https://github.com/NOAA-EMC/AQM-Workflow into develop

AnnaSmoot-NOAA committed:
add 6hr python aqm_cycled.def

AnnaSmoot-NOAA committed:
new naming aqm_cycled_def.py

BrianCurtis-NOAA committed:
Merge branch 'ecf_update' of github.com:AnnaSmoot-NOAA/AQM-Workflow into develop

AnnaSmoot-NOAA committed:
latest python files in ecf/def

BrianCurtis-NOAA committed:
Merge branch 'ecf_update' of github.com:AnnaSmoot-NOAA/AQM-Workflow into develop

AnnaSmoot-NOAA committed:
Merge branch 'NOAA-EMC:develop' into ecf_update

BrianCurtis-NOAA committed:
Merge branch 'ecf_update' of github.com:AnnaSmoot-NOAA/AQM-Workflow into develop

AnnaSmoot-NOAA committed:
removing aqm_cycled_def_v2.py

AnnaSmoot-NOAA committed:
Merge branch 'ecf_update' of github.com:AnnaSmoot-NOAA/AQM-Workflow into ecf_update

BrianCurtis-NOAA committed:
Merge branch 'ecf_update' of github.com:AnnaSmoot-NOAA/AQM-Workflow into develop

BrianCurtis-NOAA committed:
initial test yaml db file to work on coding with

AnnaSmoot-NOAA committed:
Merge pull request #1 from BrianCurtis-NOAA/ecf_update

add test db file for coding

BrianCurtis-NOAA committed:
First days work on building the workflow through python (#2)

BrianCurtis-NOAA committed:
Bring in detect_machine.py and supported_machines.yaml for machine detection (#3)

* First days work on building the workflow through python

* Detect machine using user editable supported_machines.yaml

* typehinting better and removing unused imports

AnnaSmoot-NOAA committed:
adding aqm_workflow luas and user_set py & yml

Commit:
wcoss2 updates user_set yml & py, aqm_wcoss lua, & machine yml

AnnaSmoot-NOAA committed:
Updating C6 wflow lua

AnnaSmoot-NOAA committed:
latest version of aqm_wflow_hera lua and ush/user_settings.py

AnnaSmoot-NOAA committed:
Merge branch 'ecf_update' of github.com:AnnaSmoot-NOAA/AQM-Workflow into ecf_update

AnnaSmoot-NOAA committed:
updating ush/user_set.yml & ush/user_setting.py

AnnaSmoot-NOAA committed:
latest user_settings.py & user_set.yml

AnnaSmoot-NOAA committed:
add 4 .sh scripts to ush and update user_set yml & py

AnnaSmoot-NOAA committed:
test version of python script & test scripts for reference

AnnaSmoot-NOAA committed:
updated 3 yamls & python gen script

AnnaSmoot-NOAA committed:
cleaned up yamls & ecf_gen.py for testing on other machines

AnnaSmoot-NOAA committed:
update ecf_gen.py & user_set.py & user yaml for c6

Commit:
update yamls,bash scripts in ush for WCOSS2

AnnaSmoot-NOAA committed:
adding quick fix of pack_home

AnnaSmoot-NOAA committed:
updated ecf_generate.py

AnnaSmoot-NOAA committed:
update .md files

AnnaSmoot-NOAA committed:
Update GETTING_STARTED.md

AnnaSmoot-NOAA committed:
Update GETTING_STARTED.md

AnnaSmoot-NOAA committed:
Update user_aqm.yml

AnnaSmoot-NOAA committed:
Update test_aqm.yml

AnnaSmoot-NOAA committed:
Update nco_aqm.yml

BrianCurtis-NOAA commented:
@AnnaSmoot-NOAA is this code tested on WCOSS2 to have completed the full test definition?

BrianCurtis-NOAA commented on GETTING_STARTED.md:
```diff
@@ -1,8 +1,9 @@
+ * module load build_wcoss2_intel OR module load build_gaeac6_intel
+ * module load aqm_workflow_wcoss2_intel OR module load aqm_workflow_gaeac6_intel
+ * python -c 'import ecflow'
+ * python -c 'import yaml'
```
I don't think these 2 are needed, they just check to see if they are setup in the env and work.

BrianCurtis-NOAA commented on ecf/defs/aqm_cycled_def_copy.py:
Did we need a copy?

BrianCurtis-NOAA reviewed (changes requested):
There are a lot of files in ecf/defs, can you let me know if they're all needed. Just making sure we're not bringing extra files into the repo.

In ush, the aqm_utils_python, nexus_utils dirs should be a symlink pointing to the appropriate places in sorc. Refer to the current develop branch to see how they are handled.

AnnaSmoot-NOAA committed:
Delete ecf/defs/aqm_cyc_test.py

AnnaSmoot-NOAA committed:
Delete ecf/defs/aqm_cycled_def.py

AnnaSmoot-NOAA committed:
Delete ecf/defs/aqm_cycled_def_copy.py

AnnaSmoot-NOAA committed:
Delete ecf/defs/bash_test.sh

AnnaSmoot-NOAA committed:
Delete ecf/defs/testdb.yml

AnnaSmoot-NOAA committed:
Delete ecf/defs/testdb_2.yml

AnnaSmoot-NOAA committed:
Delete ecf/defs/us_test.py

AnnaSmoot-NOAA committed:
Delete ecf/defs/user_settings.sh

AnnaSmoot-NOAA committed:
Delete ecf/defs/us_test.sh

AnnaSmoot-NOAA committed:
Delete ecf/defs/config_test.def

AnnaSmoot-NOAA committed:
Delete ecf/defs/testdb_copy.yml

AnnaSmoot-NOAA committed:
Delete ecf/defs/ecf_host_test.py

AnnaSmoot-NOAA committed:
Delete ecf/defs/machine_test.py

AnnaSmoot-NOAA committed:
Delete ecf/defs/detect_machine.sh

AnnaSmoot-NOAA committed:
Delete ecf/defs/master_def.sh

AnnaSmoot-NOAA committed:
Delete ecf/defs/module-setup.sh

AnnaSmoot-NOAA committed:
Update GETTING_STARTED.md

AnnaSmoot-NOAA commented:
@BrianCurtis-NOAA 
1. Yes, I did an E2E test on WCOSS2. 
2. I cleaned up the ecf/def/ files. 
3. I tested that the python import statements are not needed and deleted them. 
4. Fixed symlinks in ush/ for nexus_utils and aqm_utils_python. 
This completes all of the requested revisions.

AnnaSmoot-NOAA committed:
Delete ush/aqm_utils_python directory

AnnaSmoot-NOAA committed:
Delete ush/nexus_utils directory

AnnaSmoot-NOAA committed:
Create aqm_utils_python

AnnaSmoot-NOAA committed:
Delete ush/aqm_utils_python

AnnaSmoot-NOAA committed:
adding symlinks in ush/

BrianCurtis-NOAA commented:
@AnnaSmoot-NOAA has told me that her recent changes would not impact the E2E testing she performed for this earlier. So it's all approved and ready for merge.
